src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -422,15 +414,14 @@ export const createXMLHttpRequestOverride = (
         return null
       }
 
-      const headerValue = Object.entries(this.responseHeaders).reduce<
-        string | null
-      >((_, [headerName, headerValue]) => {
-        return headerName.toLowerCase() === name.toLowerCase()
```

Comment:
This actually fixes an issue: if there are multiple headers and the first one is the one you're getting, the next header in the iteration will reset the accumulated value to `null`.